sdk/@launchdarkly/observability-dotnet/src/LaunchDarkly.Observability/Observe.cs

```diff
@@ -0,0 +1,247 @@
+                new ConcurrentDictionary<string, Gauge<double>>();
+
+            public readonly ConcurrentDictionary<string, Histogram<double>> Histograms =
+                new ConcurrentDictionary<string, Histogram<double>>();
+
+            public readonly ConcurrentDictionary<string, UpDownCounter<long>> UpDownCounters =
+                new ConcurrentDictionary<string, UpDownCounter<long>>();
+
+            internal Instance(ObservabilityConfig config, ILoggerProvider loggerProvider)
+            {
+                Meter = new Meter(config.ServiceName ?? DefaultNames.MeterName,
+                    config.ServiceVersion);
+                ActivitySource = new ActivitySource(config.ServiceName ?? DefaultNames.ActivitySourceName,
+                    config.ServiceVersion);
+                if (loggerProvider != null)
+                {
+                    Logger = loggerProvider.CreateLogger(config.ServiceName ?? DefaultNames.DefaultLoggerName);
+                }
+            }
+        }
+
+        private static Instance _instance;
+
+        internal static void Initialize(ObservabilityConfig config, ILoggerProvider loggerProvider)
+        {
+            Volatile.Write(ref _instance, new Instance(config, loggerProvider));
+        }
+
+        private static Instance GetInstance()
+        {
+            return Volatile.Read(ref _instance);
+        }
+
+        private static void WithInstance(Action<Instance> action)
+        {
+            var instance = GetInstance();
+            if (instance == null)
+            {
+                // TODO: Log after PR with logger merged.
+                return;
+            }
+
+            action(instance);
+        }
+
+        private static KeyValuePair<string, object>[] ConvertToKeyValuePairs(IDictionary<string, object> dictionary)
+        {
+            if (dictionary == null || dictionary.Count == 0)
+            {
+                return null;
+            }
+
+            var result = new KeyValuePair<string, object>[dictionary.Count];
+            var index = 0;
+            foreach (var kvp in dictionary)
+            {
+                result[index] = new KeyValuePair<string, object>(kvp.Key, kvp.Value);
+                index++;
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Record an error in the active activity.
+        /// <para>
+        /// If there is not an active activity, then a new activity will be started and the error will be recorded
+        /// into the activity.
+        /// </para>
+        /// </summary>
+        /// <param name="exception">the exception to record</param>
+        /// <param name="attributes">any additional attributes to add to the exception event</param>
+        public static void RecordException(Exception exception, IDictionary<string, object> attributes = null)
+        {
+            if (exception == null)
+            {
+                // Silently return if exception is null
```

Comment:
May log here as well.

Reply:
should this throw a `ArgumentNullException` then like the others, or do nothing if it's expected to call this with a null exception sometimes?